sockeye/data_io.py

```diff
@@ -909,21 +919,30 @@ def __init__(self,
                  source_vocabs: List[Optional[str]],
                  target_vocab: Optional[str],
                  shared_vocab: bool,
-                 num_shards: int,
-                 data_statistics: DataStatistics,
-                 max_seq_len_source: int,
-                 max_seq_len_target: int) -> None:
+                 num_shards: int) -> None:
         super().__init__()
         self.sources = sources
+        self.num_source_factors = len(sources)
         self.target = target
         self.source_vocabs = source_vocabs
         self.target_vocab = target_vocab
         self.shared_vocab = shared_vocab
         self.num_shards = num_shards
+
+
+class DataConfig(config.Config):
+    """
+    Stores data statistics relevant for inference.
+    """
+
+    def __init__(self,
+                 data_statistics: DataStatistics,
+                 max_seq_len_source: int,
+                 max_seq_len_target: int) -> None:
+        super().__init__()
         self.data_statistics = data_statistics
         self.max_seq_len_source = max_seq_len_source
         self.max_seq_len_target = max_seq_len_target
-        self.num_source_factors = len(sources)
```

Comment:
wouldn't it make sense to keep the num_source_factors here, as it's something that is necessary at inference time?

Reply:
At inference time we currently get them from the embedding config:
```
@property
    def num_source_factors(self) -> int:
        """
        Returns the number of source factors of this InferenceModel (at least 1).
        """
        return self.config.config_embed_source.num_factors
```
Which one do you prefer?

Reply:
I think it is cleaner in DataConfig, but it doesn't really make a difference...

Reply:
agreed, I changed it